src/charm.py

```diff
@@ -247,44 +273,20 @@ def _on_start(self, event) -> None:
             event.defer()
             return
 
-        if self._db_initialised:
-            # The replica set should be initialised only once. Check should be
-            # external (e.g., check initialisation inside peer relation). We
-            # shouldn't rely on MongoDB response because the data directory
-            # can be corrupted.
-            return
-
         with MongoDBConnection(self.mongodb_config, "localhost", direct=True) as direct_mongo:
             if not direct_mongo.is_ready:
                 logger.debug("mongodb service is not ready yet.")
                 event.defer()
                 return
-            try:
-                logger.info("Replica Set initialization")
-                direct_mongo.init_replset()
-                logger.info("User initialization")
-                self._init_operator_user(container)
-                self._init_monitor_user()
-                logger.info("Reconcile relations")
-                self.client_relations.oversee_users(None, event)
-            except ExecError as e:
-                logger.error(
-                    "Deferring on_start: exit code: %i, stderr: %s", e.exit_code, e.stderr
-                )
-                event.defer()
-                return
-            except PyMongoError as e:
-                logger.error("Deferring on_start since: error=%r", e)
-                event.defer()
-                return
 
-            self._db_initialised = True
+        # mongod is now active
+        self.unit.status = ActiveStatus()
```

Comment:
Is it really active if we haven't run `_initialise_replica_set` yet?

Reply:
Yes, it is otherwise tests will fail